### DISCUSSION TITLE
cargo test执行报错

### DISCUSSION THREAD

**Discussion by likie0** [Tencent/tquic#138] [Q&A (问题和回答)] [answered]
在Ubuntu22系统上安装后，执行cargo test时报错，使用--nocapture参数后系统报错提示如下：
unexpected panic Some("assertion failed: tls_session_pair.client.cipher() ==\n  Some(boringssl::crypto::Algorithm::Aes128Gcm)") occured in Some(Location{ file: "src/tls/tls.rs", line: 769, col: 9}) and exit
在另一台电脑上安装，一样的系统和依赖库版本，测试并没有报错。
请问这个报错应该如何解决？

**Comment by iyangsj** [comment]
目前并未遇到或复现类似问题。

在 `src/tls/tls.rs` line 769前增加 ：
```
println!("{:?}", tls_session_pair.client.cipher());
```

然后执行以下命令，结果发出来看看
```
cargo test tls::tests::full_handshake -- --nocapture
```

如果方便，也可回复下微信号，单独拉群看看

**Reply by likie0** [reply]
![2](https://github.com/Tencent/tquic/assets/88355136/df7b144b-0736-43b1-b8c6-db0cb56166ab)

**Reply by likie0** [reply]
补充：
./tquic_server -c cert.crt -k cert.key -l 127.0.0.1:8443和./tquic_client --connect-to 127.0.0.1:8443 https://example.org
这两个命令是可以运行，得到正常结果的。

**Reply by iyangsj** [reply, marked as answer]
@likie0 感谢！已经知道大致原因，下个版本解决
可以临时删除src/tls/tls.rs: 769规避（问题在于用例本身）
